### DISCUSSION TITLE
Browser back/forward buttons cause cache and inconsistent behaviour

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [livewire/livewire#8572] [7. Bugs (archived)]
### Livewire version

3.5.0

### Laravel version

11.10.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Load up a page containing `wire:navigate`
2. Navigate forwards / backwards between pages (typically between one containing a form and another that doesn't)

### Repository or Wirebox URL

N/A

### Additional information

I'm using [`wire:navigate`](https://livewire.laravel.com/docs/navigate) in my Laravel livewire application. I have an index page and a page containing a form. When I navigate forwards & backwards between the page with the form and the page without the form using the **browser's native back & forward buttons** in Google Chrome on Windows (likely other systems too) I'm observing that the data in the form never initialises the new state correctly.

When I'm using regular links containing `wire:navigate` or navigating via the page URL, everything is fine.

I see that [this](https://github.com/livewire/livewire/issues/2603) may be related, but it's option was removed regarding back button cache.

There's still an issue around the back button & forward button.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by binaryfire** [comment]
I'm hitting the same issue. This seems related: https://github.com/livewire/livewire/discussions/6722

Did you figure out a workaround?

**Comment by dorywow** [comment]
i used 

public $search = '';

protected $queryString = [
    'search' => ['except' => ''],
];

protected $listeners = ['refreshSearchFromBrowser'];

public function updating($property): void
    {
        if (in_array($property, ['search'])) {
            $this->resetPage();
        }
    }
    
public function refreshSearchFromBrowser(string $newSearch): void
    {
        $this->search = $newSearch;
    }    
    
and this is frontend

<input type="search" wire:model.live.debounce.500ms="search" placeholder="Looking for. Škoda Octavia Scout" class="border p-2 w-full rounded" />

@push('scripts')
        <script>
            window.addEventListener('popstate', function () {
                const url = new URL(window.location.href);
                const newSearch = url.searchParams.get('search') || '';

                Livewire.find(@js($this->getId())).call('refreshSearchFromBrowser', newSearch);
            });
        </script>
    @endpush
    
Maybe it will help you

**Comment by joshhanley** [comment]
We're in the process of investigating all `wire:navigate` issues.

@sts-ryan-holton would you be able to explain what you mean by "I'm observing that the data in the form never initialises the new state correctly." If possible, please be detailed as we want to insure we have everything covered properly and we don't misunderstand what you're after/mean.

@binaryfire any chance you could write up what you're experiencing - what is happening now, and what you expect should happen instead? Screenshots/ screen captures would also be helpful. Thanks!

**Reply by binaryfire** [reply]
@joshhanley Hey mate. Two of the issues I'm hitting are covered here:

Pagination (first video): https://github.com/livewire/livewire/pull/9246#issue-2942593315 

Broken styles (and sometimes scripts) after hitting 404 or other error pages (second video): https://github.com/livewire/livewire/pull/9246#issuecomment-2848134831

**Reply by joshhanley** [reply]
@binaryfire great, thanks!

**Reply by binaryfire** [reply]
@joshhanley Just stumbled across another example in this vid. Timestamp: https://youtu.be/4hNaWxsKgvs?si=hmgeMs6GMyzX65wT&t=1401
